cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/CassandraCradleStorage.java

```diff
@@ -365,18 +365,16 @@ protected void doStoreGroupedMessageBatch(GroupedMessageBatchToStore batch, Page
 
 		try
 		{
-			messagesWorker.storeGroupedMessageBatch(batch, pageId).get();
-
 			if (settings.isStoreIndividualMessageSessions()) {
+				messagesWorker.storeGroupedMessageBatch(batch, pageId, false).get();
+
 				for (MessageBatchToStore b : batch.getSessionMessageBatches()) {
 					messagesWorker.storeMessageBatch(b, pageId).get();
 					messagesWorker.storeSession(b).get();
 					messagesWorker.storePageSession(b, pageId).get();
 				}
 			} else {
-				Map<Direction, Set<String>> sessions = getDirectionToSessionAliases(batch);
-				storeSessions(pageId, sessions).get();
-				storePageSessions(batch, sessions).get();
+				messagesWorker.storeGroupedMessageBatch(batch, pageId, true).get();
```

Comment:
As I can see `settings.isStoreIndividualMessageSessions()` is always `true` (because we throw an exception otherwise on line 347). Maybe we don't need this internal condition?

Reply:
The idea is
if isStoreIndividualMessageSessions true, doStoreMessageBatch and doStoreGroupedMessageBatch can be called
if isStoreIndividualMessageSessions false, blocks the doStoreMessageBatch call and allows doStoreGroupedMessageBatch

Reply:
Sorry, this is also the wrong method...
But I would suggest to reorganize the code like this:
```java
messagesWorker.storeGroupedMessageBatch(batch, pageId, !settings.isStoreIndividualMessageSessions()).get();

if (settings.isStoreIndividualMessageSessions()) {
	for (MessageBatchToStore b : batch.getSessionMessageBatches()) {
		messagesWorker.storeMessageBatch(b, pageId).get();
		messagesWorker.storeSession(b).get();
		messagesWorker.storePageSession(b, pageId).get();
	}
}
```